src/Tests/Tests/Ingest/ProcessorAssertions.cs

```diff
@@ -0,0 +1,345 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Elastic.Xunit.XunitPlumbing;
+using Nest;
+using Tests.Core.Client;
+using Tests.Domain;
+
+namespace Tests.Ingest
+{
+	using ProcFunc = Func<ProcessorsDescriptor, IPromise<IList<IProcessor>>>;
+	public interface IProcessorAssertion
+	{
+		object Json { get; }
+		string Key { get; }
+		IProcessor Initializer { get; }
+		ProcFunc Fluent { get; }
+	}
+	public abstract class ProcessorAssertion : IProcessorAssertion
+	{
+		public abstract string Key { get; }
+		public abstract IProcessor Initializer { get; }
+		public abstract ProcFunc Fluent { get; }
+		public abstract object Json { get; }
+	}
+
+
+	public static class ProcessorAssertions
+	{
+		public static IEnumerable<IProcessorAssertion> All =>
+			from t in typeof(ProcessorAssertions).GetNestedTypes()
+			where typeof(IProcessorAssertion).IsAssignableFrom(t) && t.IsClass
+			let a = t.GetCustomAttributes(typeof(SkipVersionAttribute)).FirstOrDefault() as SkipVersionAttribute
+			where a == null || !a.Ranges.Any(r => r.IsSatisfied(TestClient.Configuration.ElasticsearchVersion))
+			select (IProcessorAssertion) (Activator.CreateInstance(t));
+
+		public static Dictionary<string, object>[] Json =>
+			All.Select(a => new Dictionary<string, object>
+			{
+				{a.Key, a.Json}
+			}).ToArray();
+
+		public static IPromise<IList<IProcessor>> Fluent(ProcessorsDescriptor d)
+		{
+			foreach (var a in All) a.Fluent(d);
+			return d;
+		}
+
+		public static IProcessor[] Initializer => All.Select(a => a.Initializer).ToArray();
+
+		public class Append : ProcessorAssertion
```

Comment:
`AppendProcessorAssertion`, etc. ?

Reply:
I didn't bother since these are nested and will show up as 
`parent+nested.method` in test runners